test/test_core.py

```diff
@@ -3958,6 +3958,48 @@ def test_dlfcn_rtld_local(self):
 
     self.do_runf('main.c', 'main\nfunc_a\nfunc_sub\nfunc_b\nfunc_sub\ndone\n')
 
+  def test_dlfcn_with_preload(self):
+    # Create chain of dependencies and load the first libary with preload plugin.
+    # main -> libB.so -> libA.so
+    create_file('libA.c', r'''
+      #include <stdio.h>
+      int libA_fun() {
+        return 23;
+      }
+    ''')
+    self.run_process([EMCC, 'libA.c', '-sSIDE_MODULE', '-o', 'libA.so'])
+    create_file('libB.c', r'''
+      #include <stdio.h>
+      int libA_fun();
+
+      int libB_fun() {
+        return libA_fun()*2;
+      }
+    ''')
+    self.run_process([EMCC, 'libB.c', '-sSIDE_MODULE', '-o', 'libB.so', 'libA.so'])
```

Comment:
Rather than calling `run_process` directly, I think you need to either:

1. Use `self.build_dlfcn_lib`
2. Use `self.emcc`
3. Keep using self.run_process by add `self.get_emcc_args()` yourself.

Reply:
I changed it to use `build_dlfcn_lib` and `self.prep_dlfcn_main` similar to other tests.